xl-v1-common.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	slashpath "path"
```

Comment:
Refactor code shouldn't come in this PR, moving code you can do separately and squash the other commits.